syntax = "proto3";

message InputAction {
    repeated KeyboardEvent keyEvents = 1;
    sint32 mouseDx = 2;
    sint32 mouseDy = 3;
}

// // This was auto-generated
// enum KeyType {
//     FORWARD = 0;
//     BACKWARD = 1;
//     LEFT = 2;
//     RIGHT = 3;
//     JUMP = 4;
//     AUX1 = 5;
//     SNEAK = 6;
//     AUTOFORWARD = 7;
//     DIG = 8;
//     PLACE = 9;
//     ESC = 10;
//     DROP = 11;
//     INVENTORY = 12;
//     CHAT = 13;
//     CMD = 14;
//     CMD_LOCAL = 15;
//     CONSOLE = 16;
//     MINIMAP = 17;
//     FREEMOVE = 18;
//     PITCHMOVE = 19;
//     FASTMOVE = 20;
//     NOCLIP = 21;
//     HOTBAR_PREV = 22;
//     HOTBAR_NEXT = 23;
//     MUTE = 24;
//     INC_VOLUME = 25;
//     DEC_VOLUME = 26;
//     CINEMATIC = 27;
//     SCREENSHOT = 28;
//     TOGGLE_BLOCK_BOUNDS = 29;
//     TOGGLE_HUD = 30;
//     TOGGLE_CHAT = 31;
//     TOGGLE_FOG = 32;
//     TOGGLE_UPDATE_CAMERA = 33;
//     TOGGLE_DEBUG = 34;
//     TOGGLE_PROFILER = 35;
//     CAMERA_MODE = 36;
//     INCREASE_VIEWING_RANGE = 37;
//     DECREASE_VIEWING_RANGE = 38;
//     RANGESELECT = 39;
//     ZOOM = 40;
//     QUICKTUNE_NEXT = 41;
//     QUICKTUNE_PREV = 42;
//     QUICKTUNE_INC = 43;
//     QUICKTUNE_DEC = 44;
//     SLOT_1 = 45;
//     SLOT_2 = 46;
//     SLOT_3 = 47;
//     SLOT_4 = 48;
//     SLOT_5 = 49;
//     SLOT_6 = 50;
//     SLOT_7 = 51;
//     SLOT_8 = 52;
//     SLOT_9 = 53;
//     SLOT_10 = 54;
//     SLOT_11 = 55;
//     SLOT_12 = 56;
//     SLOT_13 = 57;
//     SLOT_14 = 58;
//     SLOT_15 = 59;
//     SLOT_16 = 60;
//     SLOT_17 = 61;
//     SLOT_18 = 62;
//     SLOT_19 = 63;
//     SLOT_20 = 64;
//     SLOT_21 = 65;
//     SLOT_22 = 66;
//     SLOT_23 = 67;
//     SLOT_24 = 68;
//     SLOT_25 = 69;
//     SLOT_26 = 70;
//     SLOT_27 = 71;
//     SLOT_28 = 72;
//     SLOT_29 = 73;
//     SLOT_30 = 74;
//     SLOT_31 = 75;
//     SLOT_32 = 76;
// }

enum EventType {
    PRESS = 0;
    RELEASE = 1;
}

message KeyboardEvent {
    string key  = 1;
    EventType eventType = 2;
    // KeyType key  = 1;
    // EventType eventType = 2;
}

//